packages/frontend/src/state/squeethPool/hooks.ts

```diff
@@ -0,0 +1,508 @@
+import { Contract } from 'web3-eth-contract'
+import { Token, CurrencyAmount } from '@uniswap/sdk-core'
+import { Pool, Route, Trade } from '@uniswap/v3-sdk'
+import { useUpdateAtom } from 'jotai/utils'
+import { useEffect, useCallback } from 'react'
+import { atom, useAtomValue } from 'jotai'
+import BigNumber from 'bignumber.js'
+import { ethers } from 'ethers'
+
+import useContract from '@hooks/useContract'
+import routerABI from '../../abis/swapRouter.json'
+import uniABI from '../../abis/uniswapPool.json'
+import { addressesAtom, isWethToken0Atom } from '../positions/atoms'
+import { addressAtom, networkIdAtom, web3Atom } from '../wallet/atoms'
+import useUniswapTicks from '@hooks/useUniswapTicks'
+import { DEFAULT_SLIPPAGE, OSQUEETH_DECIMALS, UNI_POOL_FEES } from '@constants/index'
+import { fromTokenAmount, parseSlippageInput } from '@utils/calculations'
+import { useETHPrice } from '@hooks/useETHPrice'
+import { useHandleTransaction } from '../wallet/hooks'
+import {
+  poolAtom,
+  wethTokenAtom,
+  squeethTokenAtom,
+  squeethInitialPriceAtom,
+  squeethPriceeAtom,
+  readyAtom,
+} from './atoms'
+
+const getImmutables = async (squeethContract: Contract) => {
+  return {
+    token0: await squeethContract?.methods.token0().call(),
+    token1: await squeethContract?.methods.token1().call(),
+    fee: await squeethContract?.methods.fee().call(),
+    tickSpacing: await squeethContract?.methods.tickSpacing().call(),
+    maxLiquidityPerTick: await squeethContract?.methods.maxLiquidityPerTick().call(),
```

Comment:
Maybe we can use Promise.all to get token, fee, tickSpacing, etc as simultaneously.